docs/book/v2/migration/preparing-for-v3.md

```diff
@@ -0,0 +1,38 @@
+# Preparing for Version 3
+
+Version 3 will introduce a number of backwards incompatible changes. This document is intended to help you prepare for these changes.
+
+## Template Resolvers
+
+In version 2, template resolvers, which all implement the method `ResolverInterface::resolve()` have varying return types along with some undocumented or inconsistent behaviour, specifically concerned with error handling, for example, when a template cannot be found, or no templates have been configured.
+
+Version 3 will solve these issues by guaranteeing a string return type from `ResolverInterface::resolve()` or throw a `\Laminas\View\Exception\ExceptionInterface`.
+
+If you use template resolvers standalone, you can prepare for this change by wrapping existing code in a try/catch
+
+```php
+// Before
+
+// Maybe null, false or string:
+return $this->resolver->resolve($name, $this->renderer);
+
+// After
+try {
+    return $this->resolver->resolve($name, $this->renderer);
+} catch (\Laminas\View\Exception\ExceptionInterface $error) {
+    return null; // Or whatever else signals a missing template in your application
+}
+```
```

Comment:
``````suggestion
#### Before

Before version 3 the return type can `null`, `false` or `string`:

```php
return $this->resolver->resolve($name, $this->renderer);
```

#### After

If the template resolvers is used as standalone, use a `try`-`catch` block to create a custom signal for a missing template in an application:

```
try {
    return $this->resolver->resolve($name, $this->renderer);
} catch (\Laminas\View\Exception\ExceptionInterface $error) {
    return null; // custom return type
}
```
``````